api/cf/cloudfoundry-types.ts

```diff
@@ -39,6 +46,45 @@ export interface GetRoleArgs {
   userGuids?: string[];
 }
 
+export interface GetServiceInstancesArgs {
+  // guids and names refer to the service instances
+  guids?: string[];
+  names?: string[];
+  organizationGuids?: string[];
+  perPage?: string;
+  servicePlanGuids?: string[];
+  servicePlanNames?: string[];
+  spaceGuids?: string[];
+  type?: Array<'managed' | 'user-provided'>;
+  // other arguments not yet implemented
+  //   fields (https://v3-apidocs.cloudfoundry.org/version/3.169.0/index.html#fields-parameter)
+  //   labelSelector (https://v3-apidocs.cloudfoundry.org/version/3.169.0/index.html#labels-and-selectors)
+  //   createdAts (timestamps and relational operators)
+  //   updatedAts (timestamps and relational operators)
+}
+
+// space and org filters do not filter out plans that are public, only those restricted to an organization
+export interface GetServicePlansArgs {
+  // guids and names refer to the service plans
+  guids?: string[];
+  names?: string[];
+  available?: boolean;
+  brokerCatalogIds?: string[];
+  organizationGuids?: string[];
+  serviceBrokerGuids?: string[];
+  serviceBrokerNames?: string[];
+  serviceOfferingGuids?: string[];
+  serviceOfferingNames?: string[];
+  serviceInstanceGuids?: string[];
+  spaceGuids?: string[];
+  include?: Array<'space.organization' | 'service_offering'>;
+  // other arguments not yet implemented
+  //   fields (https://v3-apidocs.cloudfoundry.org/version/3.169.0/index.html#fields-parameter)
+  //   labelSelector (https://v3-apidocs.cloudfoundry.org/version/3.169.0/index.html#labels-and-selectors)
+  //   createdAts (timestamps and relational operators)
+  //   updatedAts (timestamps and relational operators)
+}
+
```

Comment:
[question] I did not create objects for the API responses for the above concepts, and just used "any" for now since I was unsure on how / if / when we might actually be using them.  I'm happy to add them in if we think that would be useful, I only left them out as a shortcut.

Reply:
Went ahead and added response objects for them